store/mongo/datastore_mongo.go

```diff
@@ -306,6 +308,89 @@ func (db *DataStoreMongo) CreateIntegration(
 	return &integration, err
 }
 
+func (db *DataStoreMongo) SetIntegrationCredentials(
+	ctx context.Context,
+	integrationId uuid.UUID,
+	credentials model.Credentials,
+) error {
+	collIntegrations := db.client.Database(*db.DbName).Collection(CollNameIntegrations)
+
+	fltr := bson.D{{
+		Key:   KeyID,
+		Value: integrationId,
+	}}
+
+	update := bson.M{
+		"$set": bson.D{
+			{
+				Key:   KeyCredentials,
+				Value: credentials,
+			},
+		},
+	}
+
+	result, err := collIntegrations.UpdateOne(ctx,
+		mstore.WithTenantID(ctx, fltr),
+		update,
+	)
+	if result.MatchedCount == 0 {
+		return store.ErrObjectNotFound
+	}
+
+	return errors.Wrap(err, "mongo: failed to set integration credentials")
+}
+
+func (db *DataStoreMongo) RemoveIntegration(ctx context.Context, integrationId uuid.UUID) error {
+	collIntegrations := db.client.Database(*db.DbName).Collection(CollNameIntegrations)
+	fltr := bson.D{{
+		Key:   KeyID,
+		Value: integrationId,
+	}}
+	res, err := collIntegrations.DeleteOne(ctx, mstore.WithTenantID(ctx, fltr))
+	if err != nil {
+		return err
+	} else if res.DeletedCount == 0 {
+		return store.ErrObjectNotFound
+	}
+	return nil
+}
+
+// DoDevicesExistByIntegrationID checks if there is at least one device connected
+// with given integration ID
+func (db *DataStoreMongo) DoDevicesExistByIntegrationID(
+	ctx context.Context,
+	integrationID uuid.UUID,
+) (bool, error) {
+	var (
+		err     error
+		results = []model.Device{}
+	)
+	collDevices := db.client.Database(*db.DbName).Collection(CollNameDevices)
+	if _, err := db.GetIntegrationById(ctx, integrationID); err != nil {
+		return false, errors.Wrap(err, "error retrieving integration collection results")
+	}
```

Comment:
I don't think it's necessary to check the `integrations` collection. The following query in the `devices` collection is backed by an index.

Reply:
done, thanks!